library/TextBuilderDev/UTF16.hs

```diff
@@ -52,7 +52,7 @@ utf8CodeUnits4 byte1 byte2 byte3 byte4 case1 case2 =
   unicodeCodePoint unicode case1 case2
   where
     unicode =
-      shiftL (fromIntegral byte1 - 0xE0) 18
+      shiftL (fromIntegral byte1 - 0xF0) 18
```

Comment:
What led you to that change?

Reply:
The tests wouldn't build so investigated:

![image](https://user-images.githubusercontent.com/98815650/164649914-7dd3a496-4d24-43c3-953b-8a30bdf5b5fe.png)

A code unit sequence of length four starts with a code unit whose upper four bits are set so we have to subtract `0b11110000 = 0xF0`. That's also why I added the tests, I wanted to make sure that `Data.Text.Encoding` gets the same results.

Reply:
(Same person btw, this is my work acc)

Reply:
You're absolutely correct! Thanks for the reference and the fix. This alone is a very valuable contribution! I wonder how come nobody spotted the bug before.

Reply:
My guess is that code points outside the Basic Multilingual plane (which need 4 code units) are very very rare in practice :thinking: But maybe also not many people used the `utf8CodeUnitsN` functions.

Also you're welcome, thx for merging quickly :)